### PR TITLE
Topic 0

### PR DESCRIPTION
**Pull request by Joabulla07**
Singleton and Builder patterns

### CODE REVIEW

**Comment by Joabulla07** [comment]
Topic 1

File: topic-0/BuilderDB/src/main/java/DbConnection.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+public class DbConnection {
```
Comment (EduardoUTN) [inline topic-0/BuilderDB/src/main/java/DbConnection.java]: It's kinda Builder (but if you look close, it's a constructos passing values to another). You can implement it in different way:
* https://sourcemaking.com/design_patterns/builder/java/2 (an example using abstract class)
* https://www.geeksforgeeks.org/builder-design-pattern/ (an example using an interface)
* https://medium.com/@ajinkyabadve/builder-design-patterns-in-java-1ffb12648850 (an example using an anonymous inner class)

But, this code can be modified like the 3rd example (see above).

File: topic-0/SingletonDB/src/main/java/DatabaseConnection.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    public Connection getConnection() {
+        return connection;
+    }
+
+    public static DatabaseConnection getInstance() throws SQLException {
+        if (instance == null) {
+            instance = new DatabaseConnection();
+        } else if (instance.getConnection().isClosed()) {
```
Comment (EduardoUTN) [inline topic-0/SingletonDB/src/main/java/DatabaseConnection.java]: Both conditionals have the same side effect, you can join both using "||" operator

File: topic-1/Hello/src/main/java/Hello.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
+
+
```
Comment (EduardoUTN) [inline topic-1/Hello/src/main/java/Hello.java]: remove extra lines

File: topic-1/Hello/target/surefire-reports/Hello.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+-------------------------------------------------------------------------------
```
Comment (EduardoUTN) [inline topic-1/Hello/target/surefire-reports/Hello.txt]: add this file to .gitignore

File: topic-1/Hello/target/surefire-reports/TEST-Hello.xml [hunk 5]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (EduardoUTN) [inline topic-1/Hello/target/surefire-reports/TEST-Hello.xml]: Add the /target and childs to gitignore

**Pull request closed without merging**